src/main/java/com/example/domainmodelopdrachtspringbootles11/controller/TelevisionController.java

```diff
@@ -0,0 +1,67 @@
+package com.example.domainmodelopdrachtspringbootles11.controller;
+
+import com.example.domainmodelopdrachtspringbootles11.exceptions.RecordNotFoundException;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.ArrayList;
+
+@RestController
+public class TelevisionController {
+    ArrayList<String >televisionDataBase = new ArrayList<>();
+
+    @GetMapping("television")
+    public ResponseEntity<ArrayList<String>> getAllTelevisions() {
+        televisionDataBase.add("Sony Bravia");
+        televisionDataBase.add("Samsung QLED");
+        televisionDataBase.add("LG OLED");
+        return new ResponseEntity<>(televisionDataBase, HttpStatus.OK);
+    }
+
+    @GetMapping("television/{id}")
+    public ResponseEntity<String> getTelevisionById(@PathVariable int id) {
+        String television;
+        if (id != 1 && id != 2 && id != 3) {
+            throw new RecordNotFoundException("Opgegeven ID kan niet gevonden worden");
+        } else {
```

Comment:
Je mag nu in plaats van deze if/else if/else gewoon de repository call maken "repository.findById(id)". Daar krijg je een Optional<Television> van terug. Vervolgens moet je dan checken of er een Television in die Optional zit of niet, door optionalTelevision.isPresent() te checken. Zo ja, dan haal je de Television uit de Optional met "optionalTelevision.get()" en return je dat in responsestatus.ok, zo nee, dan return je een ResponseEntity.badRequest() of gooi je een RecordNotFoundException op.

Reply:
Gecorrigeerd. Idd geen regebruikt gemaakt van de repo.